Fix #8, #9, #10

White-Oak opened the pull request:
`amethyst new` now runs `cargo new` and appends to `Cargo.toml` an amethyst dependency.

I also updated tests.sh to track git files, maybe it should also test if `Cargo.toml` is not malformed (or should it not?).

ebkalderon commented:
@White-Oak I think it's great that you added the Git files to the testing script! Please keep it.

White-Oak commented:
@ebkalderon wow, that wrapper is really cool, thx!

White-Oak commented:
@ebkalderon 

I updated import section, but `writeln!` macro expects `std::io::Write` to be imported so I replaced `use std::io` with `use std::io::{copy, Write}`.

Other options would've been:
1. `use std::io::{self, Write}`
2. `use std::io` and add `use std::io:Write` right before `writeln!` macro.

ebkalderon commented:
Thanks, @White-Oak! That's fine. Once you run `cargo fmt` on the code, and I'll merge it.

**Edit**: Fix spelling mistake.

ebkalderon commented on src/cli/subcmds/new.rs:
```diff
@@ -25,11 +32,26 @@ pub fn execute(matches: &ArgMatches) -> Result<(), &'static str> {
+        project_path.to_string() + "Cargo.toml"
+    } else {
+        project_path.to_string() + "/Cargo.toml"
+    };
```
Oh, I also found something else that may help you eliminate this block. There is a method on paths called `join()` that might interest you: http://doc.rust-lang.org/std/path/struct.Path.html#method.join

Sorry, last-minute find.

Try this:

``` rust
let cargo_toml_path = project_path.join("Cargo.toml");
```

ebkalderon commented:
If you look at the script, it's running `amethyst new mygame`.

White-Oak commented:
@ebkalderon yeah, I mean, it doesn't panic for me...

ebkalderon commented:
@White-Oak Hmm, I need to investigate this a bit further. Let me know if you learn anything new!

**Edit**: Yeah, it works okay on my local system too. Perhaps this is something specific to Travis?
**Edit 2**: I'm restarting the Travis builds.
**Edit 3**: I've identified the problem! It's panicking only on Rust stable! The regression seems to have been fixed on beta and nightly.

ebkalderon commented:
Since Amethyst expects users to be using unstable versions of Rust, I'll create a counterpart to ebkalderon/amethyst#3 for `amethyst_tools` and disable stable builds on Travis for now. I think I'll merge this code and prepare for the 0.2.4 release. Thanks for your code and your patience! :wink:

ebkalderon commented:
@White-Oak Since Amethyst expects users to be working with unstable versions of Rust anyway, I think I'll merge this code and disable stable builds in Travis until 1.7.0 is stabilized (will open a counterpart issue for ebkalderon/amethyst#3). Thanks for your code and your patience! :wink:

ebkalderon commented:
Also @White-Oak, I just added an `AUTHORS.md` file similar to the one in Amethyst. If you would like to be publicly known as an author, just let me know and I'll add you in.

White-Oak commented:
@ebkalderon Nice! 

Yeah, I'm totally not against being in `AUTHORS.md`.

ebkalderon commented:
@White-Oak Done in 6bd3ca1!
